### ISSUE TITLE
EnumNameGeneratorType not working at all

### ISSUE THREAD

**Issue by LeonidEfremov** [RicoSuter/NSwag#3039]
Hi, @RicoSuter!

It seems like `openApiToCSharpClient.enumNameGeneratorType` (13.7.0) property not working at all... Trying to set any value - no changes at generated cs-code =(

**Comment by RicoSuter** [comment]
enumNameGeneratorType can only be change when set in-process - not via configuration (e.g. nswag.json file) because it currently cannot dynamically load types from assemblies, etc.

**Comment by RicoSuter** [comment]
To use this you'd need to write your own CLI based on the NSwag packages and then you have full flexibility...

**Comment by uncleedik** [comment]
It seems that InitCustomTypes should be called somewhere up to top, as stated in TODO item :)

**Comment by RicoSuter** [comment]
> It seems that InitCustomTypes should be called somewhere up to top, as stated in TODO item :)

It's not that easy... how would you be able to "inject" custom types into the CLI? Just calling InitCustomTypes might work when the custom type is in the CLI DLLs but probably you'd want to load your own DLL in the CLI and for that you'd also need a config to define external additional DLLs (which currently does not exist).

**Comment by uncleedik** [comment]
> 
> 
> > It seems that InitCustomTypes should be called somewhere up to top, as stated in TODO item :)
> 
> It's not that easy... how would you be able to "inject" custom types into the CLI? Just calling InitCustomTypes might work when the custom type is in the CLI DLLs but probably you'd want to load your own DLL in the CLI and for that you'd also need a config to define external additional DLLs (which currently does not exist).

Yes, i know that's not so easy, using AssemblyLoaders in any way may cause some unpredictable behaviour. Anyway, i keep in mind that this feature might be implemented, but i made a temporary console app built on NSWag code generator libraries to keep the pipeline cross-platform and simple. Have a nice day and thanks for your work!

**Comment by RicoSuter** [comment]
Ref: https://github.com/RicoSuter/NSwag/issues/924

**Comment by jhermsen** [comment]
Any plans on looking into this again?
I'm also in need of the enumNameGeneratorType.
